elastalert/alerters/smseagle.py

```diff
@@ -0,0 +1,102 @@
+        
+        self.smseagle_to = self.rule.get('smseagle_to', '')
+        self.smseagle_contacts = self.rule.get('smseagle_contacts', '')
+        self.smseagle_groups = self.rule.get('smseagle_groups', '')
+        
+        self.smseagle_duration = self.rule.get('smseagle_duration', '')
+        self.smseagle_voice_id = self.rule.get('smseagle_voice_id', '')
+    
+    def get_message_type_endpoint(self, message_type):
+        match message_type:
+            case 'ring':
+                return '/api/v2/calls/ring'
+            case 'tts':
+                return '/api/v2/calls/tts'
+            case 'tts_adv':
+                return '/api/v2/calls/tts_advanced'
+            case _:
+                return '/api/v2/messages/sms'
+
+    def alert(self, matches):
+        body = self.create_alert_body(matches)
+
+        headers = {
+            'content-type': 'application/json',
+            'access-token': self.smseagle_token
+        }
+        
+        endpoint = self.get_message_type_endpoint(self.smseagle_message_type)
+
+        payload = {
+            "text": body
+        }
+        
+        if not self.smseagle_to and not self.smseagle_contacts and not self.smseagle_groups:
+            raise EAException("Error posting SMSEagle alert: Missing recipients")
+        
+        if self.smseagle_to:
+            payload['to'] = self.smseagle_to
+            
+        if self.smseagle_contacts:
+            payload['contacts'] = self.smseagle_contacts
+                    
+        if self.smseagle_groups:
+            payload['groups'] = self.smseagle_groups
+                    
+        if self.smseagle_message_type in ['ring', 'tts', 'tts_adv']:
+            if self.smseagle_duration:
+                payload['duration'] = self.smseagle_duration
+            else:
+                payload['duration'] = 10
+            
+        if self.smseagle_message_type == 'tts_adv':
+            if self.smseagle_voice_id:
+                payload['voice_id'] = self.smseagle_voice_id
+            else:
+                payload['voice_id'] = 1
+                    
+        if self.smseagle_text:
+            payload['text'] = self.smseagle_text
+
+        for url in self.smseagle_url:
+            try:
+                response = requests.post(
+                    url+endpoint,
+                    json=payload,
+                    headers=headers
+                )
+                warnings.resetwarnings()
+                response.raise_for_status()
+                
+                elastalert_logger.debug('Response: {0}'.format(response))
+            except RequestException as e:
+                raise EAException("Error posting SMSEagle alert: %s" % e)                               
+        elastalert_logger.info("Alert '%s' sent to SMSEagle" % self.rule['name'])
+
+    def get_info(self):
+        return {'type': 'smseagle',
+        'smseagle_webhook_url': self.smseagle_url}
```

Comment:
Is this intended to use "webhook" in the key name? I didn't see that term used anywhere else in this PR.